Documentation/FrameworkOverview.md

```diff
@@ -10,198 +10,192 @@ the [Design Guidelines][].
 
-## Action
+## Actions
 
-An **Action**, represented by the [Action][] class, will do some work when
-executed with an _Input_. During or after execution, zero or more _Output_
-values are generated. Alternatively, an _Error_ may be generated.
+An **action**, represented by the [`Action`][Action] type, will do some work when
+executed with an input. While executing, zero or more output values and/or an
+error may be generated.
 
-Actions are suited to perform side-effecting work as the user interacts with
-the app.
-
-Usually the trigger for an Action is UI-driven, like when a button is
-clicked. Actions can also be automatically disabled based on a signal, and this
+Actions are useful for performing side-effecting work upon user interaction, like when a button is
+clicked. Actions can also be automatically disabled based on a [property](#properties), and this
 disabled state can be represented in a UI by disabling any controls associated
 with the action.
 
-For interaction with UIKit or AppKit GUI controls such as `NSControl` or 
-`UIControl`, RAC provides [CocoaAction][] to wrap Action with KVO or 
-Cocoa Bindings.
+For interaction with `NSControl` or `UIControl`, RAC provides the
+[`CocoaAction`][CocoaAction] type for bridging actions to Objective-C.
 
-## Property
+## Properties
 
-A **Property**, represented by the [PropertyType][Property] protocol, 
-wraps a value, allowing observation of value changes.
+A **property**, represented by the [`PropertyType`][Property] protocol,
+stores a value and notifies observers about future changes to that value.
 
 The current value of a property can be obtained from the `value` getter. The
-`producer` getter returns a [SignalProducer](#signal-producers) that will send
-the property's current value, followed by all changes over time.
-
-The `<~` operator can be used to bind properties in different ways. Note that in 
-all cases, the target has to be a `MutablePropertyType`.
-
-* `Property <~ Signal` binds the signal to the property, updating the property's 
-value to the latest value sent by the signal.
-* `Property <~ SignalProducer` creates a signal from the given producer, which will 
-immediately bind to the given property, updating the property's value to the latest 
-value sent by the signal
-* `Property <~ Property` binds the _destination_ property to the latest values 
-the _source_ property.
-
-When bridging to Objective C code (like UIKit or AppKit), `DynamicProperty` can be used to
-wrap a `dynamic` property using Key-Value-Coding and Key-Value-Observing. `DynamicProperty`
-should only be used when KVO/KVC is required by the API used (e.g. `NSOperation`), 
-`MutableProperty` should be preferred whenever possible! 
+`producer` getter returns a [signal producer](#signal-producers) that will send
+the property’s current value, followed by all changes over time.
+
+The `<~` operator can be used to bind properties in different ways. Note that in
```

Comment:
No, I don't want to confuse it with monadic bind (in case anyone hears about that and wonders how it relates).